lib/class-wp-theme-json-gutenberg.php

```diff
@@ -2093,8 +2094,9 @@ protected static function compute_style_properties( $styles, $settings = array()
 				 * whether the incoming value can be converted to a fluid value.
 				 * Values that already have a clamp() function will not pass the test,
 				 * and therefore the original $value will be returned.
+				 * Pass the current theme_json settings to override any global settings.
 				 */
-				$value = gutenberg_get_typography_font_size_value( array( 'size' => $value ) );
+				$value = gutenberg_get_typography_font_size_value( array( 'size' => $value ), $settings );
```

Comment:
Now, there's another consideration: what about preset generation?

Preset values are generated using this function too, however via a `value_func` callable property in `PRESETS_METADATA`. See:

https://github.com/WordPress/gutenberg/blob/b0162af871257438b8f994565016d7937ae010bd/lib/class-wp-theme-json-gutenberg.php#L160-L160

This callable only receives the font size value argument and not the settings:

https://github.com/WordPress/gutenberg/blob/b0162af871257438b8f994565016d7937ae010bd/lib/class-wp-theme-json-gutenberg.php#L1834-L1834

I'm not sure if it's kosher, but a follow up might have to solve this by sending `$settings` as the second arg.

```
$value      = call_user_func( $value_func, $preset, $settings );
```

Reply:
Done the above, with tests.

`Passing $settings to the callbacks defined in static::PRESETS_METADATA.`

Settings is an object, so in theory it can be extended. 

cc @oandregal for thoughts